casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -298,15 +298,24 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
     implicit val rand: Blake2b512Random = Blake2b512Random(
       DeployData.toByteArray(ProtoUtil.stripDeployData(deploy))
     )
-    // TODO: add error handling
-    for {
-      parsed    <- Interpreter[F].buildNormalizedTerm(deploy.term)
-      result    <- reducer.inj(parsed).attempt
-      phlos     <- reducer.phlo
-      oldErrors <- errorLog.readAndClearErrorVector()
-      newErrors = result.swap.toSeq.toVector
-      allErrors = oldErrors |+| newErrors
-    } yield (CostAccount.toProto(phlos) -> allErrors)
+
+    val parsingCost = accounting.parsingCost(deploy.term)
+
+    Interpreter[F].buildNormalizedTerm(deploy.term).attempt.flatMap {
+      case Right(parsed) =>
+        for {
+          result    <- reducer.inj(parsed).attempt
+          phlosLeft <- reducer.phlo
+          oldErrors <- errorLog.readAndClearErrorVector()
+          newErrors = result.swap.toSeq.toVector
+          allErrors = oldErrors |+| newErrors
+        } yield (CostAccount.toProto(phlosLeft - parsingCost) -> allErrors)
```

Comment:
Ideally we'd do `costAccounting.charge(parsingCost)` in here, but that's probably a story for a different PR. `ChargingReducer` with its `phlo` and `setPhlo` methods **MUST DIE**